Could not find a package configuration file provided by "google_cloud_cpp_rest_internal"

**Does this issue affect the google-cloud-cpp project?**
vcpkg @ [`a5d6d1`](https://github.com/microsoft/vcpkg/tree/a5d6d145164e82e67fbf91a4a30f98699d30de63)

**What component of `google-cloud-cpp` is this related to?**
```json
"dependencies": [
    {
      "name": "google-cloud-cpp",
      "default-features": true,
      "features": [
        "storage"
      ]
    }
  ]
```

**Describe the bug**
```
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
CMake Error at vcpkg/scripts/buildsystems/vcpkg.cmake:824 (_find_package):
  By not providing "Findgoogle_cloud_cpp_rest_internal.cmake" in
  CMAKE_MODULE_PATH this project has asked CMake to find a package
  configuration file provided by "google_cloud_cpp_rest_internal", but CMake
  did not find one.

  Could not find a package configuration file provided by
  "google_cloud_cpp_rest_internal" with any of the following names:

    google_cloud_cpp_rest_internalConfig.cmake
    google_cloud_cpp_rest_internal-config.cmake

  Add the installation prefix of "google_cloud_cpp_rest_internal" to
  CMAKE_PREFIX_PATH or set "google_cloud_cpp_rest_internal_DIR" to a
  directory containing one of the above files.  If
  "google_cloud_cpp_rest_internal" provides a separate development package or
  SDK, be sure it has been installed.
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.23.1/share/cmake/Modules/CMakeFindDependencyMacro.cmake:47 (find_package)
  cmake-build-debug/vcpkg_installed/x64-osx/share/google_cloud_cpp_storage/google_cloud_cpp_storage-config.cmake:16 (find_dependency)
  vcpkg/scripts/buildsystems/vcpkg.cmake:824 (_find_package)
  google_cloud_bucket_c/CMakeLists.txt:29 (find_package)


-- Configuring incomplete, errors occurred!
See also "google-cloud-bucket-c/cmake-build-debug/CMakeFiles/CMakeOutput.log".
See also "google-cloud-bucket-c/cmake-build-debug/CMakeFiles/CMakeError.log".
```

**To Reproduce**
Steps to reproduce the behavior:
1. `git clone --depth=1 https://github.com/offscale/google-cloud-bucket-c`
2. `mkdir google-cloud-bucket-c/build && cd "$_"`
3. `cmake -DCMAKE_TOOLCHAIN_FILE='[root]/vcpkg/scripts/buildsystems/vcpkg.cmake' -DCMAKE_BUILD_TYPE='Debug' ..`
4. `cmake --build .`
5. See error

Or if you don't want to use my package, then use the aforementioned `vcpkg.json` with:
```cmake
find_package(google_cloud_cpp_storage CONFIG REQUIRED)
target_link_libraries(
        "${LIBRARY_NAME}"
        PRIVATE
        "CURL::libcurl" "google-cloud-cpp::storage" "google-cloud-cpp::experimental-storage-grpc"
)
```

**Expected behavior**
No error

**Operating system:**
macOS 12.2.1 (21D62)

**What compiler and version are you using?**
```
Apple clang version 13.0.0 (clang-1300.0.29.30)
Target: x86_64-apple-darwin21.3.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

**What version of `google-cloud-cpp` are you using?**
1.40.1

**Additional context**
This was working previously

--- a/ports/google-cloud-cpp/portfile.cmake
+++ b/ports/google-cloud-cpp/portfile.cmake
@@ -47,7 +47,7 @@ foreach(feature IN LISTS FEATURES)
 endforeach()
 # These packages are automatically installed depending on what features are
 # enabled.
-foreach(suffix common googleapis grpc_utils)
+foreach(suffix common googleapis grpc_utils rest_internal)
     set(config_path "lib/cmake/google_cloud_cpp_${suffix}")
     if(NOT IS_DIRECTORY "${CURRENT_PACKAGES_DIR}/${config_path}")
         continue()

--- a/ports/google-cloud-cpp/vcpkg.json
+++ b/ports/google-cloud-cpp/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "google-cloud-cpp",
   "version": "1.40.1",
+  "port-version": 1,
   "description": "C++ Client Libraries for Google Cloud Platform APIs.",
   "homepage": "https://github.com/googleapis/google-cloud-cpp",
   "license": "Apache-2.0",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2590,7 +2590,7 @@
     },
     "google-cloud-cpp": {
       "baseline": "1.40.1",
-      "port-version": 0
+      "port-version": 1
     },
     "google-cloud-cpp-common": {
       "baseline": "alias",

--- a/versions/g-/google-cloud-cpp.json
+++ b/versions/g-/google-cloud-cpp.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "506bdfbf9040c856f8f3a4a2c183da38f9030817",
+      "version": "1.40.1",
+      "port-version": 1
+    },
+    {
       "git-tree": "2a18f7bea0f4811da422b3099c4331305cb1f617",
       "version": "1.40.1",
       "port-version": 0

